src/NitroSharp/Content/ArchiveManager.cs

```diff
@@ -0,0 +1,255 @@
+            {
+                if (head.Children == null)
+                {
+                    head.Children = new Dictionary<string, VFSNode>();
+                }
+
+                if (!head.Children.ContainsKey(part))
+                {
+                    head.Children[part] = new VFSNode();
+                }
+
+                head = head.Children[part];
+            }
+
+            if (head.MountedArchives == null)
+            {
+                head.MountedArchives = new List<IArchiveFile>();
+            }
+
+            string fullPath = Path.Combine(rootDirectory, mountPoint.ArchiveName);
+            if (File.Exists(fullPath))
+            {
+                Func<MemoryMappedFile,Encoding,IArchiveFile>[] constructors = new Func<MemoryMappedFile,Encoding,IArchiveFile>[]
+                {
+                    NPAFile.Create,
+                    AFSFile.Create,
+                };
+                MemoryMappedFile mmFile = MemoryMappedFile.CreateFromFile(fullPath, FileMode.Open, null, 0, MemoryMappedFileAccess.Read);
+                IArchiveFile? archive = null;
+                foreach (Func<MemoryMappedFile,Encoding,IArchiveFile> constructor in constructors)
+                {
+                    try
+                    {
+                        archive = constructor(mmFile, _encoding);
+                        archive.OpenArchive();
+                        break;
+                    }
+                    catch
+                    {
+                        continue;
+                    }
+                }
+                if (archive == null)
+                {
+                    throw new FileLoadException("Archive format not recognized", fullPath);
+                }
+                if (mountPoint.FileNamesIni != null)
+                {
+                    // Only AFSFile supports INI files
+                    ((AFSFile)archive).LoadFileNames(content.OpenStream(mountPoint.FileNamesIni));
+                }
+                head.MountedArchives.Add(archive);
+            }
+            else
+            {
+                // Only AFSFile supports sub-archives
+                (IArchiveFile? parentArchive, string? archivePath) = LocateFileInArchives(mountPoint.ArchiveName);
+                if (parentArchive == null || archivePath == null)
+                {
+                    throw new FileNotFoundException("Sub-archive not found", mountPoint.ArchiveName);
+                }
+                AFSFile archive = new AFSFile((AFSFile) parentArchive, archivePath, _encoding);
+                archive.OpenArchive();
+                if (mountPoint.FileNamesIni != null)
+                {
+                    archive.LoadFileNames(content.OpenStream(mountPoint.FileNamesIni));
+                }
+                head.MountedArchives.Add(archive);
+            }
+        }
+
+        private (IArchiveFile? parentArchive, string? archivePath) LocateFileInArchives(string path)
```

Comment:
``(IArchiveFile parentArchive, string archivePath)?`` seems like a better return type since they're either both null or not null.